src/Analyzers/Core/CodeFixes/RemoveUnusedParametersAndValues/AbstractRemoveUnusedValuesCodeFixProvider.cs

```diff
@@ -295,7 +295,7 @@ protected sealed override async Task FixAllAsync(Document document, ImmutableArr
                 var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
                 foreach (var diagnosticsToFix in diagnosticsGroupedByMember)
                 {
-                    var orderedDiagnostics = diagnosticsToFix.OrderBy(d => d.Location.SourceSpan.Start);
+                    var orderedDiagnostics = diagnosticsToFix.OrderByDescending(d => GetDepthOfLocation(d.Location, root));
```

Comment:
@mavasani I feel this is a common problem for CodeFixProvider (if a syntax node has multiple same diagnostics in it, and the user asks to fix all) So is there any existing solution for this? (my solution here is just a demo to what's the problem)

Reply:
it's actually normally easier to just do:

```suggestion
                    var orderedDiagnostics = diagnosticsToFix.OrderByDescending(d => d.Location.SourceSpan.Start));
```

in the case of non-contained entities, doing this in this order is fine.  in the case of contained entities, this will always process the nested element.  

we do this in a *lot* of fix all places.

Reply:
:) Happy to see it has been already a good solution for this.

Reply:
:) Great to know there is a good pattern